src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -34,6 +36,31 @@ private class CacheItem
             public TxOut[] OriginalOutputs;
         }
 
+        /// <summary>
+        /// Rewind data with its byte size that will be kept in the queue until it is persisted to a storage
+        /// </summary>
+        private class QueuedRewindData
+        {
+            /// <summary>Rewind data to be saved.</summary>
+            public RewindData RewindData { get; set; }
+
+            /// <summary>The size of the rewind data in bytes.</summary>
+            public long? DataSize { get; set; }
+
+            public static QueuedRewindData Create(RewindData rewindData)
+            {
+                long size;
+                using (var ms = new MemoryStream())
+                {
+                    var bitcoinStream = new BitcoinStream(ms, true);
+                    bitcoinStream.ReadWrite(rewindData);
```

Comment:
This is not okay. We should decide whether we want the byte size or not. I think we could be fine without the actual byte size and instead just having the number of elements that the rewind data is holding.

Or, we can go this way and really measure exactly how much data we have, but in that case we need to reuse the data when we are saving to the database. This means that the queue rewind data should actually be just byte array which is already serialised, so that we do not serialise it again when we actually do the writing to the database.

As it is right now, we do have a massive overhead of extra serialisation which is just wasted. So please decide on either not having the exact size and just using the number of elements we are holding inside or reuse the serialised data.

One could argue that in the case of reusing the serialise data, we will have overhead when reorg happens, but is that is a very rare case, it is totally fine.

Reply:
I agree. It does feel a bit heavy. I think we can go with `number of transactions to remove + number of original ones`. It will not be very accurate but probably is good enough for a flush logic. The question now is, what is a sensible threshold? I have no idea: @Aprogiena, @ferdeen  or @noescape00?